adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/JDBCSpace.java

```diff
@@ -0,0 +1,184 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.nosqlbench.adapter.jdbc;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import io.nosqlbench.api.config.standard.ConfigModel;
+import io.nosqlbench.api.config.standard.NBConfigModel;
+import io.nosqlbench.api.config.standard.NBConfiguration;
+import io.nosqlbench.api.config.standard.Param;
+import io.nosqlbench.api.errors.OpConfigError;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.sql.Connection;
+import java.util.Optional;
+
+public class JDBCSpace implements AutoCloseable {
+    private final static Logger logger = LogManager.getLogger(JDBCSpace.class);
+    private final String spaceName;
+    private HikariConfig hikariConfig;
+    private HikariDataSource hikariDataSource;
+    private Connection connection;
+
+    public JDBCSpace(String spaceName, NBConfiguration cfg) {
+        this.spaceName = spaceName;
+        this.hikariDataSource = createClient(cfg);
+    }
+
+    public Connection getConnection() {
+        return this.connection;
+    }
+
+    public HikariDataSource getHikariDataSource() {
+        return this.hikariDataSource;
+    }
+
+    private HikariDataSource createClient(NBConfiguration cfg) {
+        hikariConfig = new HikariConfig();
+
+        hikariConfig.setJdbcUrl(cfg.get("url"));
+        hikariConfig.addDataSourceProperty("serverName", cfg.get("serverName"));
+
+        Optional<String> databaseName = cfg.getOptional("databaseName");
+        if (databaseName.isPresent()) {
+            hikariConfig.addDataSourceProperty("databaseName", databaseName.get());
+        }
+
+        int portNumber = Integer.parseInt(cfg.get("portNumber"));
+        hikariConfig.addDataSourceProperty("portNumber", portNumber);
+
+        Optional<String> user = cfg.getOptional("user");
+        if (user.isPresent()) {
+            hikariConfig.setUsername(user.get());
+        }
+
+        Optional<String> password = cfg.getOptional("password");
+        if (password.isPresent()) {
+            if (user.isEmpty()) {
+                throw new OpConfigError("Both user and password options are required. Only password is supplied in this case.");
+            }
+            hikariConfig.setPassword(password.get());
+        } else {
+            if (user.isPresent()) {
+                throw new OpConfigError("Both user and password options are required. Only user is supplied in this case.");
+            }
+        }
+
+        Optional<Boolean> ssl = cfg.getOptional(Boolean.class, "ssl");
```

Comment:
In a separate request, we should see if we can configure the standard SSL provider configuration here. It uses the same config settings for all drivers and supports JVM and openssl distributions.

Reply:
Yeah, we could explore that would work after the initial stub is merged. Maybe?

Reply:
Yes